ale_linters/nim/nimcheck.vim

```diff
@@ -3,6 +3,7 @@
 
 
 function! ale_linters#nim#nimcheck#Handle(buffer, lines) abort
+    let l:buffer_filename = fnameescape(fnamemodify(bufname(a:buffer), ':p:t'))
```

Comment:
I think you can do away with `fnameescape` here and below. You only need to escape filenames if they are being used in a command line string.